staging/src/k8s.io/api/core/v1/toleration.go

```diff
@@ -51,6 +57,30 @@ func (t *Toleration) ToleratesTaint(taint *Taint) bool {
 		return t.Value == taint.Value
 	case TolerationOpExists:
 		return true
+	case TolerationOpLt, TolerationOpGt:
+		return compareNumericValues(t.Value, taint.Value, t.Operator)
+	default:
+		return false
+	}
+}
+
+// compareNumericValues performs numeric comparison between toleration and taint values
+func compareNumericValues(tolerationVal, taintVal string, op TolerationOperator) bool {
+	tVal, err := strconv.ParseInt(tolerationVal, 10, 64)
```

Comment:
ditto about the leading 0

Reply:
Actually, we shouldn't check here for the same [reason](https://github.com/kubernetes/kubernetes/pull/134665#discussion_r2475702092).